brokers/che-plugin-broker/broker.go

```diff
@@ -107,13 +107,13 @@ func (cheBroker *ChePluginBroker) PushEvents(tun *jsonrpc.Tunnel) {
 	cheBroker.Broker.PushEvents(tun, model.BrokerStatusEventType, model.BrokerResultEventType, model.BrokerLogEventType)
 }
 
-func (cheBroker *ChePluginBroker) processPlugin(meta model.PluginMeta) error {
+func (cheBroker *ChePluginBroker) ProcessPlugin(meta model.PluginMeta, onlyMetadata bool) error {
```

Comment:
Semantically, I would prefer the "default" option to be `true` instead of false (i.e. rename property to something like `copyFiles`). This is more of a personal preference though -- up to you.

Reply:
A further change here, to avoid requiring additional parameters to method calls, would be to break down the functions more directly, and have `ProcessPlugin` and `ProcessMetas` be separate entirely. The benefits there would be 
1. It seems like `onlyMetadata` applies only when the url is for an archive, which is maybe unclear
2. `onlyMetadata` causes early returns and has to be tracked across a few function calls

I also only mean this as a suggestion, though.

Reply:
I understand your remarks. Just 2 points:
- As for the `copyFiles` naming, my idea was more to point out that we might only be interested in the **metadata** part of things, just as mentioned in point 3 of this comment from @l0rd  https://github.com/eclipse/che/issues/12398#issue-398338347. But I can change it if necessary.
- As for splitting functions, I agree that some refactoring might be useful, but my idea was mainly to change the structure of the code as less as possible, knowing that a number of changes are ongoing or planned. I would have let the refactoring phase for the upcoming changes planned by @garagatyi.